x-pack/plugins/infra/public/pages/metrics/hosts_landing/index.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { EuiButton, EuiCallOut } from '@elastic/eui';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { EuiLink } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import { enableInfrastructureHostsView } from '@kbn/observability-plugin/public';
+import { HostsPage } from '../hosts';
+import { EnableHostViewPage } from './enable_host_view_page';
+
+export const HostsLandingPage = () => {
+  const kibana = useKibana();
+  const canEditAdvancedSettings = kibana.services.application?.capabilities.advancedSettings.save;
+  const [isHostViewEnabled, setIsHostViewEnabled] = useState<boolean | undefined>(undefined);
+
+  const ROLE = i18n.translate('xpack.infra.hostsLandingPage.role', {
+    defaultMessage: 'role',
+  });
+
+  useEffect(() => {
+    setIsHostViewEnabled(kibana.services.uiSettings?.get(enableInfrastructureHostsView));
+  }, [isHostViewEnabled, kibana.services.uiSettings]);
```

Comment:
**nit:** is the `kibana.services.uiSettings` a reactive state that can potentially trigger this effect or is it a mutable configuration? Also, I don't see any usage for `isHostViewEnabled` state on this effect, it can be removed.
In case `kibana.services.uiSettings` is not reactive, what's the UX scenario where you want to update this state?

Reply:
Now I am thinking that we don't need that effect at all. I was thinking at the beginning that there will be updates and it will get triggered (also was a bit different implementation before) but I tested it and it won't. So I will remove it, good catch, thanks 🙂